src/uu/touch/src/touch.rs

```diff
@@ -210,12 +213,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(|t| t.to_owned());
 
     if is_first_filename_timestamp(reference, date.as_deref(), timestamp.as_deref(), &filenames) {
-        timestamp = if filenames[0].len() == 10 {
-            Some(shr2(filenames[0]))
-        } else {
-            Some(filenames[0].to_string())
-        };
-        filenames = filenames[1..].to_vec();
+        if let Some(first_file) = filenames[0].to_str() {
```

Comment:
Can you merge those 2 if conditions into one? (forget what if let allows exactly)